File: SRP/Ejemplo2.php
<?php

class Report {
    private $title; //Atributo que almacena el titulo del reporte
    private $body; //Atributo que almacena el cuerpo del reporte

    //Constructor que inicializa los atributos del reporte
    public function __construct($title, $body) {
        $this->title = $title;
        $this->body = $body;
    }

    //Este metodo genera el reporte (Violacion del SRP)
    public function generateReport() {
        echo("Generating report...");
    }

    //Este metodo imprime el reporte (Violacion del SRP)
    public function printReport() {
        echo("Printing report: " . $this->title);
        }
}

//Se crea una instancia de la clase Report con un titulo y cuerpo
$oReport= new Report("Titulo del informe", "cuerpor del inform");

//Se llama al metodo para generar el reporte (Violacion del SRP)
$oReport->generateReport();

//Se llama al metodo para imprimir el reporte (Violacion del SRP)
$oReport->printReport();

?>

File: SRP/Ejemplo2.java
//Clase Report: Almacena los datos del reporte, cumple con SRP
public class Report {
    private String title;
    private String body;

    //Constructor que inicializa el titulo y el cuerpo del reporte
    public Report(String title, String body) {
        this.title = title;
        this.body = body;
    }

    //Metodo que devuelve el titulo del reporte
    public String getTitle() {
        return title;
    }

    //Metodo que devuelve el cuerpo del reporte
    public String getBody() {
        return body;
    }
}

//Se encarga exclusivamente de la generacion del reporte
public class ReportGenerator {
    public void generateReport(Report report) {
        // Genera el reporte en un formato específico
        System.out.println("Generating report: " + report.getTitle());
    }       
}

//Se encarga exclusivamente de imprimir el reporte
public class ReportPrinter {
    public void printReport(Report report) {
        // Imprime el reporte
        System.out.println("Printing report: " + report.getTitle());
    }
}